sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -1138,20 +1138,36 @@ def clear_state(self, window, tag):
     if not self.state[window]:
       self.state.pop(window, None)
 
-  def get_timers(self, clear=False, watermark=MAX_TIMESTAMP):
+  def get_timers(self, clear=False, watermark=MAX_TIMESTAMP, current_time=None):
+    """Gets expired timers and reports if there
+    are any realtime timers set per state.
+
+    Expiration is measured against the watermark for event-time timers,
+    and against a wall clock for processing-time timers.
+    """
     expired = []
+    has_realtime_timer = False
     for window, timers in list(self.timers.items()):
       for (name, time_domain), timestamp in list(timers.items()):
-        if timestamp <= watermark:
+        if time_domain == 'REAL_TIME':
```

Comment:
<!--new_thread; commit:92ae785f427ba708614ab0ad110b465142150610; resolved:0-->
Please use the `TimeDomain` enum (i.e. `TimeDomain.WATERMARK`, etc) here.